### PR TITLE
Add GPU runner for linux-aarch64

### PR DESCRIPTION
**Pull request by leofang**

### CODE REVIEW

**Review by ksimpson-work — APPROVED** [review]
Seems legit so long as we get the runner issues sorted, and rerun the CI before submission in case the problem spans multiple runner platforms

**Comment by leofang** [comment]
It seems on aarch64 we hit this issue... https://github.com/actions/setup-python/issues/961

File: .github/workflows/gh-build-and-test.yml [hunk 1]
```diff
@@ -76,17 +76,19 @@ jobs:
...
-    runs-on: ${{ (inputs.host-platform == 'linux-x64' && 'linux-amd64-gpu-v100-latest-1') }}
-    # TODO: use a different (nvidia?) container, or just run on bare image
+    runs-on: ${{ (inputs.host-platform == 'linux-x64' && 'linux-amd64-gpu-v100-latest-1') ||
+                 (inputs.host-platform == 'linux-aarch64' && 'linux-arm64-gpu-a100-latest-1') }}
+    # Our self-hosted runners require a container
+    # TODO: use a different (nvidia?) container
     container:
       options: -u root --security-opt seccomp=unconfined --privileged --shm-size 16g
-      image: condaforge/miniforge3:latest
+      image: ubuntu:22.04
```
Comment (leofang) [inline .github/workflows/gh-build-and-test.yml:91]: `setup-python` is being very restrictive, as it purposely locks in to Ubuntu:
- https://github.com/actions/setup-python/issues/716
- https://github.com/actions/setup-python/issues/961

(And for some reason, the `miniforge3` container can still work w/ `setup-python` on linux-64 but not on linux-aarch64...)

**Comment by leofang** [comment]
All runners are green, let me merge.

File: .github/workflows/gh-build-and-test.yml (hunk 1 above)
Comment (jakirkham) [reply .github/workflows/gh-build-and-test.yml:91]: We added ARM support earlier this year: https://github.com/conda-incubator/setup-miniconda/pull/331

So when you have a moment, could you please [raise the Miniforge GHA specific issue]( https://github.com/conda-incubator/setup-miniconda/issues/new?template=Blank+issue )

Edit: Or perhaps this is as simple as switching to `conda-incubator/setup-miniconda`

Comment (leofang) [reply .github/workflows/gh-build-and-test.yml:91]: If the question is "why don't you use conda to set up a Python environment?" the answer is it's on our TODO list: https://github.com/NVIDIA/cuda-python/issues/280. Contribution is more than welcome 😉

Comment (jakirkham) [reply .github/workflows/gh-build-and-test.yml:91]: Just saw the issue flagged above and tried to help convert this into an upstream issue or provide a path forward

Comment (leofang) [reply .github/workflows/gh-build-and-test.yml:91]: I think the two issues are disjoint.
- As the first step, we are building a test workflow mimicking pip-only use case, so we stay away from anything conda
- In the next step, we will build another workflow mimicking Conda-only use case, so the Miniforge GHA suggestion will be greatly helpful!

**Pull request merged**